static/src/javascripts/projects/common/modules/atoms/quiz.js

```diff
@@ -0,0 +1,38 @@
+define([
+    'bean',
+    'common/utils/$',
+    'common/utils/fastdom-promise'
+], function (
+    bean,
+    $,
+    fastdom
+) {
+    return {
+        handleCompletion: function () {
+            if (HTMLFormElement.prototype.checkValidity) {
+                var $quiz = $('.js-atom-quiz--instant-reveal');
+                if ($quiz.length > 0) {
+                    bean.on(document, 'click', Array.prototype.slice.call($quiz), function (e) {
+                        var quiz = e.currentTarget;
+                        if (quiz.checkValidity()) {
+                            var $bucket__message = null,
+                                bucket = $(':checked + .atom-quiz__answer__item--is-correct', quiz).length;
+                            do {
+                                $bucket__message = $('.js-atom-quiz__bucket-message--' + bucket, quiz);
+                                if ($bucket__message.length > 0) {
+                                    fastdom.write(function () {
+                                        $bucket__message.css({
+                                            'display': 'block'
+                                        });
+                                    });
+                                    break;
+                                }
+                                bucket--;
+                            } while (bucket >= 0);
```

Comment:
Ah, makes sense! Sorry I didn't notice the `break`.

How about this (ES6 speak can be converted to Lodash/ES5)? May still be misunderstanding the code, please shout if so!

``` js
var buckets = $(':checked + .atom-quiz__answer__item--is-correct', quiz);
var $bucketMessage = toArray(buckets)
    .map((bucket, index) => $('.js-atom-quiz__bucket-message--' + index, quiz)[0])
    .find(bucket => bucket);
if ($bucketMessage) {
    fastdom.write(function () {
        $bucketMessage.css({
            'display': 'block'
        });
    });
}
```